delivery_transsmart/models/res_config.py

```diff
@@ -108,47 +108,87 @@ def get_transsmart_service(self):
     def get_transsmart_carrier_tag(self):
         return self.env['ir.model.data'].get_object('delivery_transsmart', 'res_partner_category_transsmart_carrier')        
 
+    def _get_odata_filter(self, transsmart_ids):
+        """
+        assembling an odata $filter
+        http://www.odata.org/documentation/
+        odata-version-2-0/uri-conventions/
+
+        Gets all the records that their Id is not in transsmart_ids
+        """
+        _filter = ''
+        for index, code in enumerate(transsmart_ids):
+            if index < len(transsmart_ids) - 1:
+                _filter += 'Id ne {} and '.format(transsmart_ids[index])
+            else:
+                _filter += 'Id ne {}'.format(transsmart_ids[index])
+        return _filter
+
     @api.multi
     def sync_transsmart_models(self):
-        raise Warning("This option is disabled at the moment. Please update transsmart data manually or remove"
-                      "this warning from the code in the Transsmart Delivery module")
-        remote_data = self.get_transsmart_service().receive('/ServiceLevelOther')
+        # if true, fetch everything from the server and overwrite the existing
+        # values of the local records
+        # if false, set up a filter on our request and don't fetch stuff we
+        # already have.
+        update_local_data = True
         local_data = self.env['delivery.service.level'].search([])
-        local_codes = {local.code: local for local in local_data}
+        local_transsmart_ids = [local.transsmart_id for local in local_data]
+        params = {'$filter': self._get_odata_filter(local_transsmart_ids)}
+        remote_data = self.get_transsmart_service().receive(
+            '/ServiceLevelOther',
+            params=params if not update_local_data else {})
         for data in remote_data:
-            if not data['Code'] in local_codes:
+            if not data['Id'] in local_transsmart_ids:
                 self.env['delivery.service.level'].create({
                     'code': data['Code'], 
```

Comment:
And in the same spirit of DRY: use a vals dictionary for data to be used on eiither create or write.